### DISCUSSION TITLE
Difference in scoring between Scan in web interface and received email headers

### DISCUSSION THREAD

**Discussion by locojohn** [rspamd/rspamd#3671] [General]
I noticed that some of SPAM emails are not rejected by rspamd and find their way into my INBOX. When I scan message contents using rspamd web interface, I get the following results indicating total score of 17.05, which should instantly reject message:

```
DBL_SPAM (6.5) [blablabla.xyz:url]
ABUSE_SURBL (5.5) [blablabla.xyz:url]
BAYES_SPAM (5.1) [100.00%]
PREVIOUSLY_DELIVERED (-2) [my@emaildomain.com]
RECEIVED_SPAMHAUS_CSS (1) [23.228.104.214:received]
SEM_URIBL_FRESH15 (0.5) [blablabla.xyz:url]
MIME_HTML_ONLY (0.2)
ZERO_FONT (0.1) [1]
RCVD_NO_TLS_LAST (0.1)
MANY_INVISIBLE_PARTS (0.05) [1]
ARC_NA (0)
FROM_HAS_DN (0)
PRECEDENCE_BULK (0)
TO_DN_NONE (0)
RCPT_COUNT_ONE (0) [1]
DMARC_NA (0) [blablabla.xyz]
FROM_EQ_ENVFROM (0)
R_DKIM_NA (0)
MIME_TRACE (0) [0:~]
RCVD_COUNT_TWO (0) [2]
MID_RHS_MATCH_FROM (0)

Score: 17.05 / 14
```

However, when I get that same message in my INBOX, I see the following in message headers:

```
X-Spamd-Result: default: False [5.25 / 14.00];
	 ARC_NA(0.00)[];
	 BAYES_SPAM(5.10)[100.00%];
	 R_SPF_ALLOW(-0.20)[+a];
	 FROM_HAS_DN(0.00)[];
	 ZERO_FONT(0.10)[1];
	 PRECEDENCE_BULK(0.00)[];
	 TO_DN_NONE(0.00)[];
	 DMARC_NA(0.00)[blablabla.xyz];
	 RCPT_COUNT_ONE(0.00)[1];
	 MANY_INVISIBLE_PARTS(0.05)[1];
	 TO_MATCH_ENVRCPT_ALL(0.00)[];
	 MIME_HTML_ONLY(0.20)[];
	 RCVD_COUNT_ZERO(0.00)[0];
	 FROM_EQ_ENVFROM(0.00)[];
	 R_DKIM_NA(0.00)[];
	 MIME_TRACE(0.00)[0:~];
	 ASN(0.00)[asn:46573, ipnet:XX.XX.XX.0/24, country:US];
	 MID_RHS_MATCH_FROM(0.00)[]
```

Seems like some of the rules don't apply in this case, e.g. DBL_SPAM or ABUSE_SURBL. Are they not enabled by default? In that case, why is this not stated in the documentation along with information how to enable them?

I tried adding "rbl" module to options.inc:
```
filters = "chartable,dkim,rbl,regexp,fuzzy_check";
```
But that didn't work as I got an error in rspamd.log:

```
2021-03-08 13:48:57 #436504(main) <red3uk>; cfg; rspamd_init_filters: requested unknown module rbl
```

I suppose "rbl" module being a LUA module cannot be specified in "filters".  But then how can I enable it or any other LUA module and how can I verify that they are enabled in the current configuration?  I find it confusing..

**Comment by vstakhov** [comment]
Not an issue.

**Comment by vstakhov** [comment]
https://en.wikipedia.org/wiki/Time-of-check_to_time-of-use

**Comment by locojohn** [comment]
Thank you for your comments.  However, I would very much appreciate any help related to my issue.

Note: I put "enabled = true;" in local.d/rbl.conf and it didn't make a difference.

**Comment by aeigus** [comment]
Could anyone help me out to sort this issue please?  Maybe I misunderstand something, but I thought that "rbl" is enabled by default.
